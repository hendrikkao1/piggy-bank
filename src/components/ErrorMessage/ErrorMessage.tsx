import React from "react";

interface ErrorMessageProps extends React.ComponentPropsWithoutRef<"p"> {}

export const ErrorMessage = ({ children, ...rest }: ErrorMessageProps) => (
  <p className="text-xs text-red-900" role="alert" {...rest}>
    {children}
  </p>
);
