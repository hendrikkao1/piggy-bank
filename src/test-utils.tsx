import React from "react";
import {
  RenderHookOptions,
  RenderOptions,
  render,
  renderHook,
} from "@testing-library/react";
import { NextIntlClientProvider } from "next-intl";
import messages from "./messages/en-US.json";

export const AllTheProviders = ({
  children,
}: {
  children: React.ReactNode;
}) => (
  <NextIntlClientProvider locale={"en-US"} messages={messages}>
    {children}
  </NextIntlClientProvider>
);

const customRender = (ui: React.ReactElement, options: RenderOptions = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <TProps, TResult>(
  callback: (props: TProps) => TResult,
  options?: RenderHookOptions<TProps>,
) => renderHook(callback, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";

export { customRender as render, customRenderHook as renderHook };
