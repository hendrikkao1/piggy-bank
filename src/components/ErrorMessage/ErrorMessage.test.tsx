import { render } from "@/test-utils";
import { ErrorMessage } from "./ErrorMessage";

describe("ErrorMessage", () => {
  it("renders", () => {
    const { container } = render(<ErrorMessage>error</ErrorMessage>);

    expect(container).toMatchSnapshot();
  });
});
