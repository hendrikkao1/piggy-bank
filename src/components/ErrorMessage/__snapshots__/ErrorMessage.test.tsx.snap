// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage renders 1`] = `
<div>
  <p
    class="text-xs text-red-900"
    role="alert"
  >
    error
  </p>
</div>
`;
